### PR TITLE
Add `Banner` slot to Dialog + other improvements

### PR DESCRIPTION
**Pull request by langermank**
### Banner stuff

Displaying a Banner inside a Dialog is a common pattern. We should just offer a slot for it so we can control the spacing/alignment.

Confirmation dialog

### Dialog

- The size offerings for the `title` weren't exposed at the top level as they are in the Experimental version. This change to include `header_variant` should make upgrading easier
- The `padding` param was missing from the body
- Bug fix: the padding on the body was missing if the header has `divider`
- Use `IconButton` instead of custom styles (can't delete them until Experimental is upgraded in dotcom)

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: app/components/primer/alpha/dialog.pcss [hunk 1]
```diff
@@ -187,10 +209,6 @@ dialog.Overlay:not([open]) {
   &.Overlay-header--divided {
     padding-bottom: var(--stack-padding-condensed);
     box-shadow: inset 0 calc(var(--borderWidth-thin) * -1) var(--borderColor-default);
-
-    & + .Overlay-body {
```
Comment (langermank) [inline app/components/primer/alpha/dialog.pcss:191]: This has apparently been broken for awhile 😦

File: previews/primer/alpha/dialog_preview.rb [hunk 2]
```diff
@@ -222,6 +224,17 @@ def right_side(title: "Test Dialog", subtitle: nil, size: :medium, button_text:
           d.with_body { body_text }
         end
       end
+
+      # @label Confirmation Dialog
+      #
+      # @param title [String] text
+      # @param button_text [String] text
```
Comment (langermank) [inline previews/primer/alpha/dialog_preview.rb:231]:
```suggestion
      # @param button_text [String] text
      # @snapshot interactive
```

File: previews/primer/alpha/dialog_preview/with_banner.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<%= render(Primer::Alpha::Dialog.new(id: "my-dialog", title: title)) do |d| %>
+  <% d.with_show_button { button_text } %>
+  <% d.with_banner(scheme: :warning) do %>
+    <b>Warning</b>: this will also delete the temporary private fork
+  <% end %>
+  <% d.with_body { "If you're ready, publishing this advisory will also delete the temporary private fork. This action can only be undone by contacting support." } %>
+  <% d.with_footer do %>
```
Comment (langermank) [inline previews/primer/alpha/dialog_preview/with_banner.html.erb:7]:
```suggestion
  <% d.with_footer(show_divider: :true) do %>
```

File: app/components/primer/alpha/dialog.pcss [hunk 4]
```diff
@@ -110,6 +113,25 @@ dialog.Overlay:not([open]) {
...
 
+  &:has(.Overlay-header--divided) {
+    & .Overlay-body {
+      padding-top: var(--stack-padding-normal);
+    }
+  }
+
+  /* Banner */
+
+  &:has(.Banner) {
```
Comment (keithamus) [inline app/components/primer/alpha/dialog.pcss:124]: Can’t use `has` yet.

**Pull request closed without merging**